tendrl/node_agent/manager/manager.py

```diff
@@ -227,6 +228,29 @@ def on_pull(self, raw_data):
                     node_id=raw_data["node_id"],
                 )
             )
+        if "disks" in raw_data:
+            LOG.info("on_pull, Updating disks")
+            try:
+                self.etcd_client.delete(
+                    ("nodes/%s/Disks") % raw_data["node_id"], recursive=True)
+            except etcd.EtcdKeyNotFound as ex:
+                LOG.debug("Given key is not present in etcd . %s", ex)
+            disks = raw_data['disks']
+            if "disks" in disks:
+                for disk in disks['disks']:
+                    disk['node_id'] = raw_data['node_id']
+                    disk_obj = Disk(disk)
+                    disk_json = disk_obj.to_json_string()
```

Comment:
I understand you specifically convert to json and then write the Disk details to etcd. This you are doing because etcd object write does write field by field and there are multiple calls involved to etcd on that case. This used to be a performance issue as there are huge no of attributes associated with Disk object.
Just an info to others behind using this way of writing data.